crates/starknet-devnet-core/src/starknet/mod.rs

```diff
@@ -288,18 +288,26 @@ impl Starknet {
     pub(crate) fn generate_new_block(&mut self, state_diff: StateDiff) -> DevnetResult<Felt> {
         let mut new_block = self.pending_block().clone();
 
+        let new_block_number =
+            BlockNumber(new_block.block_number().0 - self.blocks.aborted_blocks.len() as u64);
+
         // set new block header
-        new_block.set_block_hash(new_block.generate_hash()?);
+        if self.config.lite_mode {
+            new_block.set_block_hash(BlockHash::from_prefixed_hex_str(&format!(
+                "0x{:x}",
+                new_block_number.0
+            ))?);
+        } else {
+            new_block.set_block_hash(new_block.generate_hash()?);
+        }
```

Comment:
Can be shortened to `new_block.set_block_hash(if self.config.lite_mode { ... } else { ... })`